src/jit/codegenxarch.cpp

```diff
@@ -2725,6 +2746,7 @@ CodeGen::genStoreRegisterReturnInLclVar(GenTreePtr treeNode)
         if (type1 != TYP_UNKNOWN)
         {
             if (structDesc.eightByteClassifications[1] == SystemVClassificationTypeIntegerReference ||
+                structDesc.eightByteClassifications[1] == SystemVClassificationTypeIntegerByReference ||
```

Comment:
Using the new IsIntegral/SseSlot methods. The code for type0 and type1 is doing different things - one is setting the firstInt/FloatUsed flags and the other is testing them.